### PR TITLE
Add command line flight recorder startup

### PR DESCRIPTION
**Pull request by stooke**
This PR adds the ability to start a Graal program and have Flight Recorder produce a JFR file with no JFR-specific code in the target source.

Delay and duration work but currently have some issues if there is still a timer running when the program exits.

### CODE REVIEW

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
+package com.oracle.svm.core.jdk.jfr.remote;
```
Comment (jiekang) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java:27]: Please add a license header for this file.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+ * Handlers the startup and teardown of JFR when run from either the command line or remote interface.
+ * Implemented by standard JFR calls as user code might do.
+ */
+
+public class JfrAutoSessionManager {
+
+    private static final String DEFAULT_JFR_FILENAME = "recording.jfr";
+    private static final String DEFAULT_JFR_CONFIG = "default";
+
+    Configuration jfrConfiguration;
```
Comment (jiekang) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java]: Can these be made private? And for jfrConfiguration, final?

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+    public static void startupHook() {
+        out("Jfr startupHook() - start");
+        out(dumpJfrOptions(", "));
+        assert instance == null;
+        if (JfrOptions.getStartRecordingAutomatically()) {
+            try {
+                // JFR-TODO fix JfrOptions to parse configuration name instead of using default "default"
+                instance = new JfrAutoSessionManager();
+                String fn = (JfrOptions.getRecordingFileName() != null && JfrOptions.getRecordingFileName().length() > 0) ? JfrOptions.getRecordingFileName() : DEFAULT_JFR_FILENAME;
```
Comment (jiekang) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java]: Can the DEFAULT_FILENAME actually be generated to include time/date or something such that its unique across multiple runs of the native image?

**Review by jiekang — CHANGES_REQUESTED** [review]
I think this included changes from #24 but merged into a different commit. I've approved and merged #24 so please rebase this on top appropriately. Thanks!

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                                stopSession(true, false);
+                            }
+                        },
+                        durationMilliseconds
+                );
+            }
+        }
+    }
+
+    void stopSession(boolean write, boolean clear) {
```
Comment (jiekang) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java:111]: The boolean write is only used in log output and the current callers always supply true. What is its purpose?

Comment (stooke) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java:111]: unimplemented (yet) flexibility in the remote control code.  I'd rather leave it in for now, if that's okay; I think the remote control PR will have a lot of discussion about what it should e like.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java (hunk 2 above)
Comment (stooke) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java]: done.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java (hunk 1 above)
Comment (stooke) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java:27]: done.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java (hunk 3 above)
Comment (stooke) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java]: done.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java (hunk 4 above)
Comment (jiekang) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java:111]: Okay, that is fine with me.

**Review by jiekang — APPROVED** [review]

**Pull request merged**